src/main/java/seedu/duke/command/AutogradeAssignmentCommand.java

```diff
@@ -0,0 +1,63 @@
+package seedu.duke.command;
+
+import seedu.duke.Module;
+import seedu.duke.Storage;
+import seedu.duke.Student;
+import seedu.duke.assignment.Answer;
+import seedu.duke.assignment.Assignment;
+import seedu.duke.assignment.McqAssignment;
+import seedu.duke.assignment.ShortAnswerAssignment;
+import seedu.duke.data.Data;
+import seedu.duke.exception.AssignmentNotFoundException;
+import seedu.duke.exception.DataFileNotFoundException;
+import seedu.duke.exception.FileFormatException;
+import seedu.duke.exception.ModuleNotFoundException;
+import seedu.duke.exception.NotAutogradableException;
+import seedu.duke.exception.NumbersMisalignException;
+import seedu.duke.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.logging.Logger;
+
+public class AutogradeAssignmentCommand extends Command {
+
+    private String moduleCode;
+    private String assignmentName;
+    private static Logger logger = Logger.getLogger(ViewAnswersCommand.class.getName());
+
+    public AutogradeAssignmentCommand(String moduleCode, String assignmentName) {
+        this.moduleCode = moduleCode;
+        this.assignmentName = assignmentName;
+    }
+
+    @Override
+    public void execute(Data data, Ui ui, Storage storage) throws ModuleNotFoundException,
+            AssignmentNotFoundException, DataFileNotFoundException, NumbersMisalignException,
+            NotAutogradableException, FileFormatException {
+        Module module = data.find(moduleCode);
+        if (module == null) {
+            throw new ModuleNotFoundException();
+        }
+        assert module != null : "module should not be null";
+        Assignment assignment = module.findAssignment(assignmentName);
+        if (assignment == null) {
+            throw new AssignmentNotFoundException();
+        }
+        assert assignment != null : "assignment should not be null";
+        Answer answer = storage.loadAnswer(assignmentName, moduleCode);
+        if (assignment.getAnswers() == null) {
+            assignment.setAnswers(answer);
+        }
```

Comment:
I see that this implementation only sets the answers, if assignment.getAnswers() == null. Maybe you could consider changing it so that we set the answers every time the answers are loaded from text file? This way the answers will update in case the text file was changed.